src/inputcommand.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module typing/inputcommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command/command';
+import ChangeBuffer from './changebuffer';
+
+/**
+ * The input command. Used by the {@link module:typing/input~Input input feature} to handle typing.
+ *
+ * @extends core.command.Command
+ */
+export default class InputCommand extends Command {
+	/**
+	 * Creates an instance of the command.
+	 *
+	 * @param {module:core/editor/editor~Editor} editor
+	 */
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * Typing's change buffer used to group subsequent changes into batches.
+		 *
+		 * @protected
+		 * @member {typing.ChangeBuffer} #_buffer
+		 */
+		this._buffer = new ChangeBuffer( editor.document, editor.config.get( 'typing.undoStep' ) || 20 );
```

Comment:
This should be given to the command as an option, not taken by it from the editor. A class like a command should depend on the world in as little details as possible.